source/code/include/playfab/PlayFabPlatformUtils.h

```diff
@@ -7,53 +7,125 @@
+        return Clock::from_time_t(input);
     }
 
-    inline std::string LocalTimeTToUtcString(time_t now)
+    inline tm TimeTToUtcTm(time_t input)
     {
         tm timeInfo;
+#if defined(PLAYFAB_PLATFORM_WINDOWS) || defined(PLAYFAB_PLATFORM_XBOX) || defined(PLAYFAB_PLATFORM_PLAYSTATION)
+#define gmtime_r gmtime_s
+#endif
+        gmtime_r(&input, &timeInfo);
+        return timeInfo;
+    }
+
+    inline time_t UtcTmToTimeT(tm input)
+    {
+#if defined(PLAYFAB_PLATFORM_PLAYSTATION)
+        return mktime(&input);
+#else
 #if defined(PLAYFAB_PLATFORM_WINDOWS) || defined(PLAYFAB_PLATFORM_XBOX)
-        gmtime_s(&timeInfo, &now);
-#elif defined(PLAYFAB_PLATFORM_LINUX) || defined(PLAYFAB_PLATFORM_IOS) || defined(PLAYFAB_PLATFORM_ANDROID) || defined(PLAYFAB_PLATFORM_PLAYSTATION)
-        timeInfo = *gmtime(&now);
+#define timegm _mkgmtime
 #endif
-        char buff[64];
-        strftime(buff, 64, "%Y-%m-%dT%H:%M:%S.000Z", &timeInfo);
+        return timegm(&input);
+#endif
+    }
 
-        return buff;
+    inline tm TimePointToUtcTm(TimePoint input)
+    {
+        return TimeTToUtcTm(Clock::to_time_t(input));
     }
 
-    inline std::string LocalTimePointToUtcString(TimePoint now)
+    inline TimePoint UtcTmToTimePoint(tm input)
     {
-        return LocalTimeTToUtcString(std::chrono::system_clock::to_time_t(now));
+        return TimeTToTimePoint(UtcTmToTimeT(input));
     }
 
-    inline time_t UtcStringToLocalTimeT(const std::string& utcString)
+    // Get Time now - Platform dependent granularity (granularity: upto 1 second, accuracy within a few seconds)
+    inline TimePoint GetTimePointNow()
     {
-        time_t output;
-        tm timeStruct = {};
+        // The conversion is mostly to ensure consistent behavior among all platforms
+        return std::chrono::time_point_cast<std::chrono::seconds>(Clock::now());
+    }
 
-        std::istringstream iss(utcString);
-        iss >> std::get_time(&timeStruct, "%Y-%m-%dT%T");
-        timeStruct.tm_isdst = 0;  // 0 means "not in DST" PlayFab assumes UTC/Zulu always
-#if defined(PLAYFAB_PLATFORM_PLAYSTATION)
-        output = mktime(&timeStruct);
-#elif defined(PLAYFAB_PLATFORM_IOS) || defined(PLAYFAB_PLATFORM_ANDROID) || defined(PLAYFAB_PLATFORM_LINUX)
-        gmtime_r(&output, &timeStruct);
+    inline time_t GetTimeTNow()
+    {
+        return TimePointToTimeT(GetTimePointNow());
+    }
+
+    // Get a tick count that represents now in milliseconds (not useful for absolute time)
+    inline long GetMilliTicks()
+    {
+#if defined(PLAYFAB_PLATFORM_SWITCH)
+        static_assert("You must request the Nintendo specific XPlat SDK from PlayFab support.");
 #else
-        output = _mkgmtime(&timeStruct);
+        auto msClock = std::chrono::time_point_cast<std::chrono::milliseconds>(Clock::now());
+        return static_cast<long>(msClock.time_since_epoch().count());
 #endif
+    }
+
+    // Time Serialization
+    inline std::string UtcTmToIso8601String(tm input)
```

Comment:
I think it's idiomatic to avoid taking `tm` by value since it's a decent sized struct. recommend using a const-ref